include/zephyr/llext/loader.h

```diff
@@ -73,6 +73,7 @@ struct llext_loader {
 	elf_shdr_t sects[LLEXT_MEM_COUNT];
 	enum llext_mem *sect_map;
 	uint32_t sect_cnt;
+	size_t prog_data_size;
```

Comment:
Since this is being used outside of this code, it deserves a doc comment explaining what its for and being placed outside the doc ignore block